### DISCUSSION TITLE
environment variable not working

### DISCUSSION THREAD

**Discussion by syedyousafraza** [Synthetixio/synpress#272] [Q&A] [answered]
can you please tell me where I should add the environment variables (directory?) like PRIVATE_KEY, SECRET_WORDS etc for metamask wallet? right now it's not recognizing the one I added in my cypress.json file. I also tried without env variable using ;
setupMetamask(secretWordsOrPrivateKey, network, password), for example: setupMetamask('word1, word2, etc..', 'mainnet', 'password').
but This is also not working i am getting error; 
Error: Please provide SECRET_WORDS or PRIVATE_KEY environment variable
    at Object.<anonymous> (E:\cypressZedrun\node_modules\←[4m@synthetixio←[24m\synpress\synpress.js:13:9)

**Comment by drptbl** [comment, marked as answer]
@syedyousafraza `cypress.env` is not `process.env`, so it's not working.

It's explained in the README:
https://github.com/Synthetixio/synpress#-important

You have many different options:
1. You can use env vars as a prefix before your command, for example: `ENV_VAR=VALUE synpress run`
Lets say that you have `npm run test` which runs `synpress run`. Then you can also run `ENV_VAR=VALUE npm run test`.
2. You can use `dotenv` (check `preload` section):
https://www.npmjs.com/package/dotenv
Create new `.env` file with variables and then run `synpress run` additionally requiring your `.env` file.
3. You can specify env variables with `export ENV_VAR=VALUE` before running tests.

.. and many more different ways. You can also refer to synpress usage examples which are also mentioned in README:
https://github.com/Synthetixio/synpress#%EF%B8%8F-synpress

Cheers,
Jakub.

**Reply by dion49** [reply]
hi @drptbl I'm running into a similar issue. When I run ` synpress help`  I get the same message; Error: Please provide SECRET_WORDS or PRIVATE_KEY environment variable. I have ` .env`  file at my root with all the vars set

` "dependencies": {
    "@synthetixio/synpress": "^1.1.1",
    "dotenv": "^10.0.0"
  },`
 ` 
  "devDependencies": {
    "@cypress/code-coverage": "^3.9.12",
    "chai-sorted": "^0.2.0",
    "cypress": "^9.2.0",
    "cypress-metamask": "^1.0.5-development",
    "synthetix-js": "^2.56.1"
  },` 
` 
  "scripts": {
    "test": "e2e",
    "cy:open": "CYPRESS_REMOTE_DEBUGGING_PORT=9222 cypress open"
  },`

**Reply by PascalMayr** [reply]
I've created a .env file and used dotenv to preload all the envs defined in that file. The following works for me:
`dotenv -f ./tests/e2e/.env npx synpress run`

**Reply by lucianoacsilva** [reply]
@PascalMayr , have you run dotenv as a global dependency?

**Reply by PascalMayr** [reply]
@lucianoacsilva yes

**Reply by Loque18** [reply]
I can't get this to work :(

**Comment by steezeburger** [comment]
I had to set the path correctly in `playwright.config.ts` for the envfile. You can actually see a working example here https://github.com/astriaorg/web-apps/pull/197/files

The playwright and synpress commands seemingly run the config file first. I thought I was to load the env in the wallet setup file, but that file gets transpiled to js and called from `.cache-synpress` I believe.
